### ISSUE TITLE
Installing spree 0.70.1

### ISSUE THREAD

**Issue by Arrumaco** [citrus/spree_variant_options#9]
Hi, I'm developing an store in spree 0.70.1 and want to include the functionality of spree_variant_options. I followed the instructions to install this extension on 0.70.1 as showed in the page: 

http://spreecommerce.com/extensions/170-spree-variant-options 

for the Dev Releases - 0.70.x edge and, added this line to my Gemfile

gem 'spree_variant_options', :git => 'git://raw.github.com/citrus/spree_variant_options.git', :branch => '0.70.x' 

but when I run the 'bundle install', it says:

> bundle install
> Updating git://raw.github.com/citrus/spree_variant_options.git
> fatal: ambiguous argument '0.70.x': unknown revision or path not in the working tree.
> Use '--' to separate paths from revisions
> fatal: ambiguous argument '0.70.x': unknown revision or path not in the working tree.
> Use '--' to separate paths from revisions
> Fetching source index for http://rubygems.org/
> Could not find gem 'spree_variant_options (>= 0)' in git://raw.github.com/citrus/spree_variant_options.git (at 0.70.x).
> Source does not contain any versions of 'spree_variant_options (>= 0)'

Could you help me noting what is my mistake?

Thanks a lot

**Comment by citrus** [comment]
Hi Arrumaco, I'm still in the process of making this 0.70.x compatible.. Hopefully later today I'll push the 0.70.x branch to github...

**Comment by Arrumaco** [comment]
Great!.. I'll wait to give it a try. 

Agustín Botana
Enviado HTC/Windows Mobile

----- Mensaje original -----
De: Spencer Steffen reply@reply.github.com
Enviado: 29 de Diciembre de 2011 02:54 p.m.
Para: Arrumaco agustin.botana@gmail.com
Asunto: Re: [spree_variant_options] Installing spree 0.70.1 (#9)

Hi Arrumaco, I'm still in the process of making this 0.70.x compatible.. Hopefully later today I'll push the 0.70.x branch to github...

---

**Comment by citrus** [comment]
Hi Agustín,

I just pushed the 0.70.x commits to the master branch.. see if installing it like this works now:

```
gem "spree_variant_options", :git => "git://github.com/citrus/spree_variant_options.git", :branch => "master"
```

Let me know if you run into any issues! 

Thanks,

-Spencer

**Comment by Arrumaco** [comment]
Hi Spencer, here my finds with the new master branch,
Everything was ok with the gem installation, no problem at all.
Running in a demo store using the spree demo data, when I select the Ruby
on Rails Baseball Jersey before displaying the options it reached an error:
NoMethodError in Products#show

Showing
C:/Ruby192/lib/ruby/gems/1.9.1/gems/spree_core-0.70.1/app/views/products/_cart_form.html.erb
where line #19 raised:

undefined method `grouped_option_values' for #Product:0x6f6caf8

Extracted source (around line #19):

16:     <h2><%= t('variants') %></h2>
17:
18:     <% index = 0 %>
19:     <% @product.grouped_option_values.each do |type, values| %>
20:       <div id="<%= dom_id(type) %>" class="variant-options index-<%=
index %>">
21:         <h3 class="variant-option-type"><%= type.presentation %></h3>
22:         <ul class="variant-option-values">

Trace of template inclusion:
C:/Ruby192/lib/ruby/gems/1.9.1/gems/spree_core-0.70.1/app/views/products/show.html.erb

I copied the two files: option_value and product decorators to my working
store and it worked fine (it seems that don't find the 'decorators' files
from the gem folder - probably is a problem with my store I'll try it again
with a new one).

With the two files copied in my store folders, it showed just fine, but I
don't see the change in the picture as I select a jersey color.. It work
well showing the rigth price for the option selected but doesn't work
changing the image displayed.

I'll keeping testing and even build a new demo store with variant_options
gem. Any notice from you would be appreciatte.

Thanks a lot.

2011/12/29 Spencer Steffen <
reply@reply.github.com

**Comment by citrus** [comment]
That seems like a strange issue.. maybe a windows bug? Do other extensions with decorators work for you?

**Comment by Arrumaco** [comment]
Yes it does. 

I see in the dev log many "deface" overrides both from my copy as from spree_promo.

BTW I opened a new issue with this same text as I saw this "closed". sorry. I just closed it.

**Comment by citrus** [comment]
I'm having trouble duplicating this issue... can you paste your Gemfile please?

**Comment by Arrumaco** [comment]
```
source 'http://rubygems.org'

gem 'rails', '3.1.1'

# Bundle edge Rails instead:
# gem 'rails',     :git => 'git://github.com/rails/rails.git'

gem 'sqlite3'
gem 'rack', '1.3.3'
gem 'mongrel', '1.2.0.pre2'


# Gems used only for assets and not required
# in production environments by default.
group :assets do
  gem 'sass-rails',   '~> 3.1.4'
  gem 'coffee-rails', '~> 3.1.1'
  gem 'uglifier', '>= 1.0.3'
end

gem 'jquery-rails'

# To use ActiveModel has_secure_password
# gem 'bcrypt-ruby', '~> 3.0.0'

# Use unicorn as the web server
# gem 'unicorn'

# Deploy with Capistrano
# gem 'capistrano'

# To use debugger
# gem 'ruby-debug19', :require => 'ruby-debug'

group :test do
  # Pretty printed test output
  gem 'turn', :require => false
end
gem 'spree', '0.70.3'
# gem 'spree_blue_theme', :git => 'git://github.com/spree/spree_blue_theme.git', :branch => '0-70-stable'
gem "spree_variant_options", :git => "git://github.com/citrus/spree_variant_options.git", :branch => "master"   
```

**Comment by Arrumaco** [comment]
source 'http://rubygems.org'

gem 'rails', '3.1.1'

# Bundle edge Rails instead:

# gem 'rails',     :git => 'git://github.com/rails/rails.git'

gem 'sqlite3'
gem 'rack', '1.3.3'
gem 'mongrel', '1.2.0.pre2'

# Gems used only for assets and not required

# in production environments by default.

group :assets do
  gem 'sass-rails',   '~> 3.1.4'
  gem 'coffee-rails', '~> 3.1.1'
  gem 'uglifier', '>= 1.0.3'
end

gem 'jquery-rails'

# To use ActiveModel has_secure_password

# gem 'bcrypt-ruby', '~> 3.0.0'

# Use unicorn as the web server

# gem 'unicorn'

# Deploy with Capistrano

# gem 'capistrano'

# To use debugger

# gem 'ruby-debug19', :require => 'ruby-debug'

group :test do
  # Pretty printed test output
  gem 'turn', :require => false
end
gem 'spree', '0.70.3'

# gem 'spree_blue_theme', :git => 'git://github.com/spree/spree_blue_theme.git', :branch => '0-70-stable'

gem "spree_variant_options", :git => "git://github.com/citrus/spree_variant_options.git", :branch => "master"

AND the Bundle list....

Gems included by the bundle:
- actionmailer (3.1.1)
- actionpack (3.1.1)
- activemerchant (1.17.0)
- activemodel (3.1.1)
- activerecord (3.1.1)
- activeresource (3.1.1)
- activesupport (3.1.1)
- acts_as_list (0.1.4)
- ansi (1.4.1)
- arel (2.2.1)
- bcrypt-ruby (3.0.1)
- braintree (2.13.1)
- builder (3.0.0)
- bundler (1.0.15)
- cancan (1.6.5)
- cocaine (0.2.1)
- coffee-rails (3.1.1)
- coffee-script (2.2.0)
- coffee-script-source (1.2.0)
- deface (0.7.2)
- devise (1.4.7)
- erubis (2.7.0)
- execjs (1.2.13)
- faker (1.0.0)
- gem_plugin (0.2.3)
- highline (1.6.2)
- hike (1.2.1)
- i18n (0.6.0)
- jquery-rails (1.0.19)
- json (1.6.4)
- kaminari (0.13.0)
- mail (2.3.0)
- meta_search (1.1.1)
- mime-types (1.17.2)
- mongrel (1.2.0.pre2)
- multi_json (1.0.4)
- nested_set (1.6.8)
- nokogiri (1.5.0)
- orm_adapter (0.0.5)
- paperclip (2.4.1)
- polyamorous (0.5.0)
- polyglot (0.3.3)
- rack (1.3.3)
- rack-cache (1.1)
- rack-mount (0.8.3)
- rack-ssl (1.3.2)
- rack-test (0.6.1)
- rails (3.1.1)
- railties (3.1.1)
- rake (0.9.2.2)
- rd_find_by_param (0.1.1)
- rd_resource_controller (1.0.1)
- rdoc (3.12)
- sass (3.1.12)
- sass-rails (3.1.5)
- spree (0.70.3)
- spree_api (0.70.3)
- spree_auth (0.70.3)
- spree_core (0.70.3)
- spree_dash (0.70.3)
- spree_promo (0.70.3)
- spree_sample (0.70.3)
- spree_variant_options (0.2.0 cfc949a)
- sprockets (2.0.3)
- sqlite3 (1.3.5)
- state_machine (1.0.1)
- stringex (1.0.3)
- thor (0.14.6)
- tilt (1.3.3)
- treetop (1.4.10)
- turn (0.8.3)
- tzinfo (0.3.31)
- uglifier (1.2.1)
- warden (1.0.6)

**Comment by citrus** [comment]
I just pushed commit d7c89f343a0a71f251d4558c1319caffcaf228ea that should fix this problem. Let me know if it works!

**Comment by Arrumaco** [comment]
Yes it works!... Will you put it in a new commit?

I'll keep trying it to migrate my data to the new features... any problem (I wish is not), I'll let you know

Amazing job.  Thanks a lot, and....  Happy new year

**Comment by citrus** [comment]
You're welcome! Happy new year to you as well.

I just released [v0.3.0](https://rubygems.org/gems/spree_variant_options) to rubygems. Now you can include spree_variant_options in your Gemfile like so: 

```
gem "spree_variant_options", "~> 0.3.0"
```

Enjoy!

-Spencer
